### DISCUSSION TITLE
Conversion to poetry ?

### DISCUSSION THREAD

**Discussion by spapas** [getredash/redash#6480] [Q&A] [answered]
Hello, I was using this project from latest master and it worked fine with the requirements.txt files.

However recently the project was switched (for no real reason) to a non-official python tool (poetry). This makes it very difficult for me to update my dependencies to latest versions. 

Can somebody help me with that? Is it possible to add a `requirements.txt` with the current versions of dependencies so I can properly install them using `pip install -r requirements.txt` ? Please notice I am not familiar with poetry and I'd rather not use nor learn a non-official tool.

Thank you for any help

**Comment by justinclift** [comment, marked as answer]
Sorry, but we've officially moved the project to using Poetry for our Python dependency management.

> Please notice I am not familiar with poetry and I'd rather not use nor learn a non-official tool.

Poetry is new to me too, and (so far) doesn't seem too bad.

It's pretty easy to install:

```
$ pip3 install poetry
```

And installing the Redash dependencies then becomes:

```
$ poetry install --only main,all_ds,dev
```

I understand you're reluctant.  Poetry is widely used by Python developers though, and seems like it'll be useful for us. :smile:

**Reply by spapas** [reply]
Thank you for the tips, I was able to install the dependencies using your recommendation!

**Reply by justinclift** [reply]
Awesome, glad to help. :smile:

**Comment by guidopetri** [comment]
The main reason to switch to poetry was actually to better version pin dependencies. With only requirements files, this was nigh-impossible, and would often break our application even without a change to the requirements file.

You should still be able to install anything you like with pip on top of poetry, they're fully compatible.
